designer/client/section-create.js

```diff
@@ -1,62 +1,83 @@
 import React from 'react'
 import { clone } from 'digital-form-builder-model/lib/helpers'
+import { camelCase } from './helpers'
 
 class SectionCreate extends React.Component {
   state = {}
 
-  onSubmit = e => {
+  async onSubmit (e) {
     e.preventDefault()
-    const form = e.target
-    const formData = new window.FormData(form)
-    const name = formData.get('name').trim()
-    const title = formData.get('title').trim()
+    const { name, title, generatedName } = this.state
     const { data } = this.props
     const copy = clone(data)
 
-    const section = { name, title }
-    copy.sections.push(section)
+    const updated = copy.addSection(name || generatedName, title.trim())
 
-    data.save(copy)
-      .then(data => {
-        console.log(data)
-        this.props.onCreate({ data })
-      })
-      .catch(err => {
-        console.error(err)
-      })
+    try {
+      const savedData = await data.save(updated)
+      this.props.onCreate(savedData)
+    } catch (err) {
+      console.error(err)
```

Comment:
what happens when it can't save?

Reply:
sure, let's chat about it - I think this should probably be a generic mechanism handled e.g. in data.save / something similar